specs/MultiProfile.md

```diff
@@ -0,0 +1,240 @@
+  [propget] HRESULT ProfileName([out, retval] LPWSTR* value);
+  /// Sets the `ProfileName` property.
+  [propput] HRESULT ProfileName([in] LPCWSTR value);
+
+  /// `InPrivateModeEnabled` property is to enable/disable InPrivate mode.
+  [propget] HRESULT InPrivateModeEnabled([out, retval] BOOL* enabled);
+  /// Sets the `InPrivateModeEnabled` property.
+  [propput] HRESULT InPrivateModeEnabled([in] BOOL enabled);
+}
+
+[uuid(57FD205C-39D5-4BA1-8E7B-3E53C323EA87), object, pointer_default(unique)]
+interface ICoreWebView2Environment5 : IUnknown
+{
+  /// Create a new ICoreWebView2ControllerOptions to be passed as a parameter of
+  /// CreateCoreWebView2ControllerWithOptions and CreateCoreWebView2CompositionControllerWithOptions.
+  HRESULT CreateCoreWebView2ControllerOptions(
+      [in] LPCWSTR profileName,
+      [in] BOOL inPrivateModeEnabled,
+      [out, retval] ICoreWebView2ControllerOptions** options);
+
+  /// Create a new WebView with options.
+  HRESULT CreateCoreWebView2ControllerWithOptions(
+      [in] HWND parentWindow,
+      [in] ICoreWebView2ControllerOptions* options,
+      [in] ICoreWebView2CreateCoreWebView2ControllerCompletedHandler* handler);
+
+  /// Create a new WebView in visual hosting mode with options.
+  HRESULT CreateCoreWebView2CompositionControllerWithOptions(
+      [in] HWND parentWindow,
+      [in] ICoreWebView2ControllerOptions* options,
+      [in] ICoreWebView2CreateCoreWebView2CompositionControllerCompletedHandler* handler);
+}
+
+[uuid(6E5CE5F0-16E6-4A05-97D8-4E256B3EB609), object, pointer_default(unique)]
+interface ICoreWebView2_7 : IUnknown {
+  /// The associated `ICoreWebView2Profile` object.
+  [propget] HRESULT Profile([out, retval] ICoreWebView2Profile** profile);
+}
+
+[uuid(3B9A2AF2-E703-4C81-9D25-FCE44312E960), object, pointer_default(unique)]
+interface ICoreWebView2Profile : IUnknown {
+  /// Name of the profile.
+  [propget] HRESULT ProfileName([out, retval] LPWSTR* value);
+
+  /// InPrivate mode is enabled or not.
+  [propget] HRESULT InPrivateModeEnabled([out, retval] BOOL* enabled);
+
+  /// Full path of the profile directory.
+  [propget] HRESULT ProfilePath([out, retval] LPWSTR* value);
+
+  // TODO: All profile-wide operations/settings will be put below in the future.
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2ControllerOptions;
+    runtimeclass CoreWebView2Environment;
+    runtimeclass CoreWebView2;
+    runtimeclass CoreWebView2Profile;
+
+    runtimeclass CoreWebView2ControllerOptions
+    {        
+        String ProfileName { get; set; };
+
+        Int32 InPrivateModeEnabled { get; set; };
+    }
+    
+    runtimeclass CoreWebView2Environment
+    {
+        // ...
+    
+        CoreWebView2ControllerOptions CreateCoreWebView2ControllerOptions(
+            String ProfileName, Int32 InPrivateModeEnabled);
```

Comment:
This too should be a Boolean not Int32. Prefixing this with `is` again may make this work better by the tool.

Reply:
Prefixing this with `is` does not seem to change the `Int32` to `Boolean`. I noticed that the same thing happens for other APIs in Microsoft.Web.WebView2.Core.idl, for instance:

>             CoreWebView2ContextMenuItem CreateContextMenuItem(String Label, String ShortcutKeyDescription, Windows.Storage.Streams.IRandomAccessStream iconStream, CoreWebView2ContextMenuItemKind Kind, Int32 IsEnabled, Int32 IsChecked);

Reply:
Looks like ContextMenu API changed and no longer has this parameter. Examining WV2Property.cs I realize now that's not enough since the generated code won't know to convert the bool to int32

Reply:
Done. Changed to use Boolean.